src/OSSupport/File.cpp

```diff
@@ -692,11 +692,10 @@ AString cFile::GetExecutableExt(void)
 
 
 
-int cFile::vPrintf(const char * a_Fmt, fmt::printf_args a_ArgList)
+int cFile::vPrintf(const char * a_Format, fmt::printf_args a_ArgList)
 {
-	fmt::memory_buffer Buffer;
-	fmt::printf(Buffer, fmt::to_string_view(a_Fmt), a_ArgList);
-	return Write(Buffer.data(), Buffer.size());
+	AString buf = ::vPrintf(a_Format, a_ArgList);
+	return Write(buf.c_str(), buf.length());
```

Comment:
Revert this.

Reply:
Please explain your reasoning.

1. Is there any evidence to suggest that there is any performance improvement, which will offset the deviation from writing obvious, maintainable code?
2. I received a warning that printf into a buffer was deprecated in a newer fmt.

Reply:
>Is there any evidence to suggest that there is any performance improvement, which will offset the deviation from writing obvious, maintainable code?

It is trivially a performance improvement if you look at the implementation of `fmt::vsprintf`:
https://github.com/fmtlib/fmt/blob/d0623de510c64234dfccbe62dc3f77a1852b1d52/include/fmt/printf.h#L628-L630

It always writes into `fmt::memory_buffer` first and then copies into a new string. So, this code saves 1 memory allocation/deallocation and 1 string copy for every write to the file.

>I received a warning that printf into a buffer was deprecated in a newer fmt.

As in the linked code, it's been renamed to `vprintf`.